pkg/operators/v1alpha1/catalogsource_types.go

```diff
@@ -92,6 +96,30 @@ type UpdateStrategy struct {
 	*RegistryPoll `json:"registryPoll,omitempty"`
 }
 
+// UnmarshalJSON implements the encoding/json.Unmarshaler interface.
+// UnmarshalJSON checks to see the provided polling interval is valid. If so, use it, if not set it to the default interval.
+func (u *UpdateStrategy) UnmarshalJSON(data []byte) error {
+	var updateStrategy map[string]map[string]string
+	if err := json.Unmarshal(data, &updateStrategy); err != nil {
+		return err
+	}
+	if updateStrategy == nil {
+		return nil
+	}
```

Comment:
Any idea on when we'd encounter the updateStrategy map being nil after the unmarshaling process?

Reply:
My reasoning was the updateStrategy field is optional on the spec and is a pointer type, so if it's not listed on the spec unmarshaling would return an empty map

Reply:
Actually, I think the unmarshaling would not occur at all, since the JSON key is missing from the spec if updateStrategy is not set, so this check might not be very useful. Still prevents indexing into a nil map though.